src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -8575,17 +8574,64 @@ Strings StorageReplicatedMergeTree::getZeroCopyPartPath(
     return res;
 }
 
+void StorageReplicatedMergeTree::addZeroCopyLock(const String & part_name, const DiskPtr & disk)
```

Comment:
I do not like the name. Because we don't create a lock, we create a watch :)
Maybe `watchZeroCopyLock` or something